### PR TITLE
Fix lineinfile to insert line when pattern exists elsewhere in the file.

### LINKED ISSUE #28721: lineinfile with_items does not insert value of line if it occurred before
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

lineinfile in combination with_items

##### ANSIBLE VERSION

```
ansible 2.3.2.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### SUMMARY
Ansible does not insert line if the identical value has been used before in loop, but only if the value of line is different.

##### STEPS TO REPRODUCE
```yaml
---
- hosts: localhost
    
  tasks:
    - name: create test file
      copy:
        dest: /tmp/example.txt
        content: |
          1
          
          2
          
          3
          
          4
          

    - name: broken lineinfile
      lineinfile:
        path: /tmp/example.txt
        insertafter: "{{ item.regex }}"
        line: "{{ item.replace }}"
        unsafe_writes: yes
      with_items:
        - { regex: '1', replace: 'bar' }
        - { regex: '2', replace: 'bar' }
        - { regex: '3', replace: 'br' }
        - { regex: '4', replace: 'bar' }

```

##### EXPECTED RESULTS
```
1
bar

2
bar

3
br

4
bar
```

##### ACTUAL RESULTS
```
1
bar

2

3
br

4
```

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

The insertion logic was being skipped if the pattern already existed in the file. It would get inserted the first time but skipped on subsequent iterations. It does not have anything to do with looping as the original issue stated, but rather the existence of the pattern in the file.

Also add integration tests.

Fixes #28721

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME
`lineinfile.py`

##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure on centos6: `failure: template error while templating string: no filter named 'map'`

**Comment by maxamillion** [comment]
>The insertion logic was being skipped if the pattern already existed in the file. It would get
> inserted the first time but skipped on subsequent iterations.

I thought that was the expected behavior, or am I missing something here?

**Comment by samdoran** [comment]
@maxamillion Sorry for the unclear explanation. If you were using `insertbefore` or `insertafter`, and the regexp happened to exist somewhere in the file but not with the before/after pattern, the insertion would be skipped. This is different than replacing a line. It's the before/after part that had the bug.

File: test/integration/targets/lineinfile/vars/main.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    replace: 'bar'
+
+  - regex: '2'
+    replace: 'bar'
+
+  - regex: '3'
+    replace: 'bar'
+
+  - regex: '4'
+    replace: 'bar'
```
Comment (maxamillion) [inline test/integration/targets/lineinfile/vars/main.yml:12]: Needs a newline

**Comment by maxamillion** [comment]
@samdoran other than the missing newline, this looks good :+1:

**Pull request merged**